src/plugins/d3/renderer/components/Chart.tsx

```diff
@@ -36,7 +36,10 @@ export const Chart = ({width, height, data}: Props) => {
     const svgRef = React.createRef<SVGSVGElement>();
     const hasAxisRelatedSeries = series.data.some(isAxisRelatedSeries);
     const {chartHovered, handleMouseEnter, handleMouseLeave} = useChartEvents();
-    const {chart, legend, title, tooltip, xAxis, yAxis} = useChartOptions(data);
+    const {chart, legend, title, tooltip, xAxis, yAxis} = useChartOptions({
+        ...data,
+        hasAxisRelatedSeries,
```

Comment:
Perhaps there is no need to pass hasAxisRelatedSeries? It can be calculated from data